applications/stream-applications-core/stream-applications-security-common/src/main/java/org/springframework/cloud/stream/app/security/common/AppStarterWebFluxSecurityAutoConfiguration.java

```diff
@@ -49,19 +61,61 @@
 @EnableConfigurationProperties(AppStarterWebSecurityAutoConfigurationProperties.class)
 public class AppStarterWebFluxSecurityAutoConfiguration {
 
+	@Autowired
+	private WebEndpointProperties webEndpointProperties;
+
+	@Bean
+	@ConditionalOnMissingBean(UserDetailsService.class)
+	public MapReactiveUserDetailsService userDetailsService(SecurityProperties securityProperties,
+			AppStarterWebSecurityAutoConfigurationProperties streamAppsecurityProperties) {
+		UserDetails primaryUser = User.builder()
+				.username(securityProperties.getUser().getName())
+				.password(securityProperties.getUser().getPassword())
+				.passwordEncoder(PasswordEncoderFactories.createDelegatingPasswordEncoder()::encode)
+				.roles(securityProperties.getUser().getRoles()
+						.toArray(new String[securityProperties.getUser().getRoles().size()]))
+				.build();
+
+		if (StringUtils.hasText(streamAppsecurityProperties.getAdminPassword()) &&
+				StringUtils.hasText(streamAppsecurityProperties.getAdminUser())) {
+			UserDetails user = User.builder()
+					.username(streamAppsecurityProperties.getAdminUser())
+					.password(streamAppsecurityProperties.getAdminPassword())
+					.roles("ADMIN")
+					.passwordEncoder(PasswordEncoderFactories.createDelegatingPasswordEncoder()::encode)
+					.build();
+			return new MapReactiveUserDetailsService(primaryUser, user);
+		}
+		return new MapReactiveUserDetailsService(primaryUser);
+	}
+
 	@Bean
 	public SecurityWebFilterChain springSecurityFilterChain(ServerHttpSecurity http,
 			AppStarterWebSecurityAutoConfigurationProperties securityProperties) {
+		String postToActuatorPath = webEndpointProperties.getBasePath() + "/**";
 		if (!securityProperties.isCsrfEnabled()) {
 			http.csrf().disable();
 		}
+		else {
+			/*
+			 * See https://stackoverflow.com/questions/51079564/spring-security-antmatchers-not-being-
+			 * applied-on-post-requests-and-only-works-wi/51088555
+			 */
+			http.csrf().requireCsrfProtectionMatcher(exchange -> new AntPathMatcher()
+					.match(postToActuatorPath,
+							exchange.getRequest().getPath().value())
+									? ServerWebExchangeMatcher.MatchResult.notMatch()
+									: ServerWebExchangeMatcher.MatchResult.match());
+		}
 		if (!securityProperties.isEnabled()) {
 			http.authorizeExchange()
 					.anyExchange()
 					.permitAll();
 		}
 		else {
-			http.authorizeExchange().pathMatchers("/actuator", "/actuator/health", "/actuator/info", "/actuator/bindings")
+			http.authorizeExchange()
+					.pathMatchers(HttpMethod.POST, postToActuatorPath).hasRole("ADMIN")
+					.pathMatchers("/actuator", "/actuator/health", "/actuator/info", "/actuator/bindings")
```

Comment:
Don't we need to use the same `EndpointRequest.to("health", "info", "bindings")` over here, too?
I guess that `webEndpointProperties.getBasePath()` could be used otherwise, but I believe to keep it as an `/actuator` prefix is wrong.

Reply:
This is a good point. The imperative version does not have the "actuator" paths. It would be a simple change to bring this inline w/ the mvc version. The api does require a string path and we can inject in PathMappedEnpdoints bean do something like this: 
```java
pathMappedEndpoints.getPath(EndpointId.of("health"))
```
to get each path.